### PR TITLE
Send RewardChainBlockUnfinished to third-party harvesters

### PR DESCRIPTION
**Pull request by haorldbchi**
`RewardChainBlockUnfinished` is now also sent to third-party harvesters along with the rest of source signing data. See #16717

### CODE REVIEW

**Comment by AmineKhaldi** [comment]
The remaining CI failures will be addressed after PR #17445 lands.

File: chia/farmer/farmer_api.py
Comment (arvidn) [inline chia/farmer/farmer_api.py:590]: the `message_data` field is suppose to be identical to the previous logic, right?

File: chia/farmer/farmer_api.py
Comment (arvidn) [inline chia/farmer/farmer_api.py:578]: previously you, effectively, had an assert ensuring `full_node_request.foliage_transaction_block_data` is `not None` (line 579). if that condition truly holds, you could assert and simplify this a bit not scrapping the `None` case

**Review by arvidn — COMMENTED** [review]
lgtm. I'll let other people review it too, before approving

File: chia/farmer/farmer_api.py
Comment (haorldbchi) [reply chia/farmer/farmer_api.py:578]: The assert was for `foliage_block_data` being `not None`, whenever we had a `full_node_request.foliage_transaction_block_data` present. Meaning `foliage_block_data` was always supposed to be present when `full_node_request.foliage_transaction_block_data` was. I simplified it by just doing it in the first branch when we check for `foliage_block_data`. Should be the same effect:

If `foliage_transaction_block_data` is present, then `foliage_block_data` must have also been present. In this case we already know it's present since we're doing the work in the same branch

File: chia/farmer/farmer_api.py
Comment (haorldbchi) [reply chia/farmer/farmer_api.py:590]: Yes, should be the same logic, just less lines of code

**Review by arvidn — DISMISSED** [review]

**Comment by DaOneLuna** [comment]
When I made these changes to my Node I also had to increase the rate limit for ProtocolMessageTypes.request_signed_values, otherwise the requests will fail for non local connections. 

ProtocolMessageTypes.request_signed_values: RLSettings(100, 2 * 1024),

Not at my desk right now but I can make these changes shortly

**Comment by DaOneLuna** [comment]
> When I made these changes to my Node I also had to increase the rate limit for ProtocolMessageTypes.request_signed_values, otherwise the requests will fail for non local connections.
> 
> ProtocolMessageTypes.request_signed_values: RLSettings(100, 2 * 1024),
> 
> Not at my desk right now but I can make these changes shortly

https://github.com/Chia-Network/chia-blockchain/pull/17481

**Review by aqk — DISMISSED** [review]

**Comment by aqk** [comment]
Amine, make sure to look at https://github.com/Chia-Network/chia-blockchain/pull/17481/files

**Comment by harold-b** [comment]
> When I made these changes to my Node I also had to increase the rate limit for ProtocolMessageTypes.request_signed_values, otherwise the requests will fail for non local connections.
> 
> ProtocolMessageTypes.request_signed_values: RLSettings(100, 2 * 1024),
> 
> Not at my desk right now but I can make these changes shortly

Thanks for catching this and making the appropriate PR, Luna

**Comment by arvidn** [comment]
let's hold off merging until https://github.com/Chia-Network/chia-blockchain/pull/17481 lands (which is targeting this PR)

**Comment by arvidn** [comment]
@haorldbchi I take it we may want to bump those 2k up to 10k

**Comment by haorldbchi** [comment]
> @haorldbchi I take it we may want to bump those 2k up to 10k

Agreed

File: chia/server/rate_limit_numbers.py
Comment (haorldbchi) [inline chia/server/rate_limit_numbers.py:1]: @arvidn

**Review by arvidn — APPROVED** [review]

**Pull request merged**
